lib/open_api/docs/doc_v2.rb

```diff
@@ -12,7 +12,19 @@ def version
       end
 
       def definitions
-        @definitions ||= content["definitions"]
+        @definitions ||= substitute_regexes(content["definitions"])
+      end
+
+      def parameters
+        @parameters ||= substitute_regexes(content["parameters"])
+      end
+
+      def expanded_definition(key)
+        definitions[key].tap do |i|
+          i["properties"] = i["properties"].each_with_object({}) do |(k, v), hash|
```

Comment:
This feels like it could be

```ruby
i["properties"] = substitute_references(i["properties"])
```

and then it can just reuse the Hash logic below.